src/main/java/com/puppycrawl/tools/checkstyle/PackageObjectFactory.java

```diff
@@ -121,38 +128,46 @@ public PackageObjectFactory(String packageName, ClassLoader moduleClassLoader) {
      */
     @Override
     public Object createModule(String name) throws CheckstyleException {
-        Object instance = createObjectFromMap(name);
+        final String nameCheck = name + CHECK_SUFFIX;
+        Object instance = createObject(name);
```

Comment:
Currently, I re-manage the order to what you stated and add the skipping when there are periods in the name.

> pure createObject(name) should probably be used last when user supplies full package as name

I don't know the situation in production, but in most UTs the name is a full qualified one. And thats why I put the "full package test" at first in my previous commit. 

In each UT we would instantiate a new `PackageObjectFactory` and call `createModule` with a full qualified name.
P.S. invoking stack:
[finishLocalSetup:419, Checker](https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/Checker.java#L419)
[configure:167, AutomaticBean](https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/api/AutomaticBean.java#L167)
[createChecker:69, BaseCheckTestSupport](https://github.com/checkstyle/checkstyle/blob/master/src/test/java/com/puppycrawl/tools/checkstyle/BaseCheckTestSupport.java#L69)

The 3rd party scan is time consuming. If we use your order without skipping, then we would do step 1, 2, 3 each time and it slows down test quite a lot.
Skipping 1-2 could help.